frontend/src/types/utils.d.ts

```diff
@@ -0,0 +1,3 @@
+type WithRequired<T, K extends keyof T> = T & {
+  [P in K]-?: T[K];
+};
```

Comment:
Not sure I can find if it's used anywhere?

Reply:
dang. that's just something I played with, forgot to remove it out. 🙏